### DISCUSSION TITLE
[docker] How to change the port?

### DISCUSSION THREAD

**Discussion by seanap** [searxng/searxng#1442] [Q&A] [answered]
Hello, I already have services running on 80 and port 8080.  How can I change the port so I can go to `my.local.ip.addr:1234` to access searx?

I have tried to set the `SEARXNG_HOSTNAME=my.local.ip.addr:1234` in the .env file but that didn't help

I have tried to set the port in the settings.yaml under the server section.

I have added `-BIND_ADDRESS:0.0.0.0:1234` in the environment section of the searxng in the docker-compose.yaml

When doing a docker-compose up I get an error for searxng `Bind for 0.0.0.0:8080 failed: port is already allocated`

Any one know what I'm missing? Thanks!

**Comment by MrHappy** [comment]
Change the port configuration in the docker-compose file...

**Reply by seanap** [reply]
In the docker-compose.yaml under `ports:` I have it set to `- "127.0.0.1:8080:1234`, and I've also changed it to `- 8080:1234` but that didn't do anything thing, how should I change that?

**Reply by Kurizek27** [reply]
why the dots? also it just broke completely. now it throws out random bs from python

**Comment by MrHappy** [comment, marked as answer]
You should change the 8080:1234 to 8081:8080 i.e..
And change caddy (I think) to connect to port 8081. (I don't use caddy, so not sure)

**Reply by seanap** [reply]
Thank You! That did it I had them the wrong way

**Comment by MrHappy** [comment]
No problem. Glad you got it sorted ;)

**Comment by quadcom** [comment]
What if you are not running the container in bridge mode? I have a dedicated IP assigned and the container is stuck at port 8080. Running this on Unraid. Could this be down to the Unraid docker template config?
